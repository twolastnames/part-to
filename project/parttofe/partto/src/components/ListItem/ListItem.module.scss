@import "../../App.scss";

.precursor {
  animation: popout var(--animation-tempo);
}

.listItem {
  display: flex;
  height: 100%;
  padding: var(--gap-xsmall);
  overflow: hidden;
  gap: var(--gap-medium);
}

.description {
  height: 100%;
  animation: fadein var(--animation-tempo);
}
