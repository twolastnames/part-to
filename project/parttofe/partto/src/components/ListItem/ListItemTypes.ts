import { ReactNode } from "react";

export interface ListItemProps {
  precursor?: ReactNode;
  description: ReactNode;
  trailer?: ReactNode;
}
