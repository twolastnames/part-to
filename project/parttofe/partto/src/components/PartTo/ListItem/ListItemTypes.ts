import { PartToId } from "../../../api/sharedschemas";

export interface ListItemProps {
  partTo: PartToId;
}
