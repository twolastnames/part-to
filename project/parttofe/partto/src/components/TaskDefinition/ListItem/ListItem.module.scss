@import "../../../App.scss";

.listItem {
  display: flex;
  height: 100%;
  padding: var(--gap-xsmall);
  overflow: hidden;
}

.description {
  height: 100%;
}
