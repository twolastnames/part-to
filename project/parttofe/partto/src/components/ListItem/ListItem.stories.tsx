import React from "react";
import type { Meta, StoryObj } from "@storybook/react";

import { ListItem } from "./ListItem";
import { ShellProvider } from "../../providers/ShellProvider";

const meta: Meta<typeof ListItem> = {
  component: ListItem,
  decorators: (Story) => (
    <ShellProvider>
      <Story />
    </ShellProvider>
  ),
};
export default meta;

type Story = StoryObj<typeof ListItem>;

export const Simple: Story = {
  args: {},
};
