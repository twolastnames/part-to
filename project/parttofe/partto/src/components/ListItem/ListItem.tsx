import React from "react";

import classes from "./ListItem.module.scss";
import { ListItemProps } from "./ListItemTypes";

export function ListItem({ precursor, description, trailer }: ListItemProps) {
  return (
    <div className={classes.listItem} data-testid="ListItem">
      {precursor && <div className={classes.precursor}>{precursor}</div>}
      <div className={classes.description}>{description}</div>
      {trailer && <div className={classes.trailer}>{trailer}</div>}
    </div>
  );
}
