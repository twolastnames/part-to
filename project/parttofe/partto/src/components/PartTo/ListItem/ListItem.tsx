import React from "react";

import { ListItemProps } from "./ListItemTypes";
import { ListItem as GenericListItem } from "../../ListItem/ListItem";
import { useParttoGet } from "../../../api/parttoget";
import { Spinner } from "../../Spinner/Spinner";
import { Recipe } from "../../Icon/Icon";

export function ListItem({ partTo }: ListItemProps) {
  const response = useParttoGet({ partTo });
  return (
    <Spinner responses={[response]}>
      <GenericListItem
        precursor={<Recipe />}
        description={response.data?.name}
      />
    </Spinner>
  );
}
