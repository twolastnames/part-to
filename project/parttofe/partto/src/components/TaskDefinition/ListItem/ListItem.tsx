import React from "react";

import { ListItemProps } from "./ListItemTypes";
import { Spinner } from "../../Spinner/Spinner";
import { useRunGet } from "../../../api/runget";
import { Icon } from "../Icon/Icon";
import { useTaskGet } from "../../../api/taskget";
import { ListItem as GenericListItem } from "../../ListItem/ListItem";

export function ListItem({ task, runState, iconClassSets }: ListItemProps) {
  const taskResponse = useTaskGet({ task });
  const response = useRunGet({ runState });

  const iconClasses = response.data?.timers.imminent
    .map((timer) => timer.task)
    .includes(task)
    ? iconClassSets.imminent
    : response?.data?.duties.includes(task)
      ? iconClassSets.duty
      : iconClassSets.task;

  return (
    <Spinner responses={[response]}>
      <GenericListItem
        precursor={<Icon classNames={iconClasses} />}
        description={taskResponse.data?.description}
      />
    </Spinner>
  );
}
