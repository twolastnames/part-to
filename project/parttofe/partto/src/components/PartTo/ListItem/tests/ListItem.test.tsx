import React from "react";
import { expect, test } from "@jest/globals";
import { render, screen, waitFor } from "@testing-library/react";
import { ShellProvider } from "../../../../providers/ShellProvider";
import { ListItem } from "../ListItem";
import partTo1 from "../../../../mocks/partTo1.json";

test("snapshot", async () => {
  fetchMock.mockResponse((request: Request) => {
    if (request.url.includes("/api/partto/")) {
      return Promise.resolve(JSON.stringify(partTo1));
    }
    return Promise.reject("");
  });
  render(
    <ShellProvider>
      <ListItem partTo="partTo1" />
    </ShellProvider>,
  );
  await waitFor(() => {
  screen.getByTestId("ListItem");
  })
  const component = screen.getByTestId("ListItem");
  expect(component).toMatchSnapshot();
});
