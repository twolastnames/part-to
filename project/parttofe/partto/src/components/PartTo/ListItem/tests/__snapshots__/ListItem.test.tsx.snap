// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  class="listItem"
  data-testid="ListItem"
>
  <div
    class="precursor"
  >
    <svg
      class="tabler-icon tabler-icon-bowl-spoon "
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M4 11h16a1 1 0 0 1 1 1v.5c0 1.5 -2.517 5.573 -4 6.5v1a1 1 0 0 1 -1 1h-8a1 1 0 0 1 -1 -1v-1c-1.687 -1.054 -4 -5 -4 -6.5v-.5a1 1 0 0 1 1 -1z"
      />
      <path
        d="M8 7c1.657 0 3 -.895 3 -2s-1.343 -2 -3 -2s-3 .895 -3 2s1.343 2 3 2"
      />
      <path
        d="M11 5h9"
      />
    </svg>
  </div>
  <div
    class="description"
  >
    Bavarian Pot Roast
  </div>
</div>
`;
